docs/contributing.rst

```diff
@@ -267,6 +267,17 @@ Please:
 .. _PEP 8 - Style Guide for Python Code:
   https://www.python.org/dev/peps/pep-0008
 
+Submitting a pull request
+=========================
+Steps:
+
+1. Write your code!
+2. Make sure your environment is set up properly and that you're in your virtualenv (this is a **very important** step).
+3. Run ``./pep8.travis.sh`` to do a cursory check of your code style. Fix any errors.
+4. Run ``tox -e lint`` to check for pylint errors. Fix any errors.
+5. Run ``tox`` to run the unit tests. Fix any errors.
+6. Run the integration tests, see `integration`_.
```

Comment:
I didn't see anywhere it's a "pre-submission" checklist. Section says "Submitting a pull request" and keeping an eye on Travis should be a part of this task. A lot of people actually don't check it and leave PR as is, while they could just look at the logs and fix. This could save a bit of a dev time.